src/OpenLoco/src/Ui/ScrollView.cpp

```diff
@@ -297,12 +297,74 @@ namespace OpenLoco::Ui::ScrollView
     // 0x004CA1ED
     void updateThumbs(Window* window, WidgetIndex_t widgetIndex)
     {
-        registers regs;
+        const auto& widget = window->widgets[widgetIndex];
+        auto& scrollArea = window->scrollAreas[window->getScrollDataIndex(widgetIndex)];
 
-        regs.esi = X86Pointer(window);
-        regs.ebx = window->getScrollDataIndex(widgetIndex) * sizeof(ScrollArea);
-        regs.edi = X86Pointer(&window->widgets[widgetIndex]);
-        call(0x4CA1ED, regs);
+        if (scrollArea.hasFlags(ScrollFlags::hscrollbarVisible))
+        {
+            int32_t viewSize = widget.width() - 22;
+            if (scrollArea.hasFlags(ScrollFlags::vscrollbarVisible))
+                viewSize -= 11;
+
+            int32_t x = scrollArea.contentOffsetX * viewSize;
+            if (scrollArea.contentWidth != 0)
+                x /= scrollArea.contentWidth;
+
+            scrollArea.hThumbLeft = x + 11;
+
+            x = widget.width() - 3;
+            if (scrollArea.hasFlags(ScrollFlags::vscrollbarVisible))
+                x -= 11;
+
+            x += scrollArea.contentOffsetX;
+            if (scrollArea.contentWidth != 0)
+                x = (x * viewSize) / scrollArea.contentWidth;
+
+            x += 11;
+            viewSize += 10;
+            scrollArea.hThumbRight = std::min(x, viewSize);
+
+            if (scrollArea.hThumbRight - scrollArea.hThumbLeft < 20)
```

Comment:
what is this new bit of code for? I assume its from openrct2?

Reply:
https://github.com/OpenRCT2/OpenRCT2/pull/4025 found it. Perhaps we add a comment that this sets a minimum size for the thumb.

Reply:
So i guess you could update the changelog as well.